packages/yoroi-extension/chrome/extension/ergo-connector/utils.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import type { Tx } from './types';
+import type { TokenRow } from '../../../app/api/ada/lib/storage/database/primitives/tables';
+import { Logger } from '../../../app/utils/logging';
+
+function parseEIP0004Data(input: any): ?string {
+  // https://github.com/ergoplatform/eips/blob/master/eip-0004.md
+  // format is: 0e + vlq(len(body as bytes)) + body (as bytes formatted in hex)
+  // where body is a utf8 string
+  // and the vlq encoding is also byte-padded e.g. 01 instead of 1
+  if (typeof input !== 'string' || !input.startsWith('0e') || input.length < 4) {
+    return null
+  }
+  let body = input.slice(2);
+  let len = 0;
+  let readNext = true;
+  do {
+    const lenChunk = parseInt(body.slice(0, 2), 16);
+    body = body.slice(2);
+    if (isNaN(lenChunk)) {
+      return null;
+    }
+    readNext = (lenChunk & 0x80) !== 0;
+    len = (128 * len) + (lenChunk & 0x7F);
+  } while (readNext);
+  if (2 * len > body.length) {
+    Logger.info(`vlq decode trailing data: ${body.slice(2 * len)}`);
+  }
+  if (2 * len < body.length) {
+    return null;
+  }
+  return Buffer.from(body.slice(0, 2 * len), 'hex').toString('utf8');
+}
+
+// you should not be able to mint more than 1
+export function mintedTokenInfo(tx: Tx): $ReadOnly<TokenRow>[] {
+  const tokens = []
+  for (const output of tx.outputs) {
+    const name = parseEIP0004Data(output.additionalRegisters.R4);
+    const description = parseEIP0004Data(output.additionalRegisters.R5);
+    const decimals = parseEIP0004Data(output.additionalRegisters.R6);
+    if (name != null && description != null && decimals != null) {
+      tokens.push({
+        TokenId: 0,
```

Comment:
I don't think we use the digest/networkid/tokenid in the tx signing page. I could try and figure out the data for them but we only use this for tx signing.

Should we bother getting this info? I think we can get the network id from looking at which addresses are used in the tx but it's a bit indirect and I'd need to use the rust module, or we could pass that in. I'm not sure where to get the digest from. The tokenId thing afaik is something tracked just within Yoroi so I don't think we can easily create something valid here.

We also have both a ticker and a longName here but there is only one name in the EIP-0004 spec so I'm not sure where we get those from.

Reply:
I wouldn't spend time on this until we absolutely need it. We'll put a note about it and do later when actually needed.